orttraining/orttraining/training_ops/cuda/activation/bias_gelu_grad_impl.cu

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "orttraining/training_ops/cuda/activation/bias_gelu_grad_impl.h"
+
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "core/providers/cuda/shared_inc/fast_divmod.h"
+#include "orttraining/training_ops/cpu/activation/gelu_computation_mode.h"
+#include "orttraining/training_ops/cuda/activation/gelu_grad_impl_common.cuh"
+
+namespace onnxruntime {
+namespace cuda {
+
+template <typename T, typename GeluComputationMode, int num_consecutive_elements_per_group, int num_groups_per_thread>
+__global__ void BiasGeluGradDxKernel(int64_t bias_size, const T* dY, const T* X, const T* B, T* dX) {
+  const int64_t input_base_idx = bias_size * blockIdx.x + num_consecutive_elements_per_group * threadIdx.x;
+  const int64_t bias_base_idx = num_consecutive_elements_per_group * threadIdx.x;
+  const int64_t group_stride = num_consecutive_elements_per_group * blockDim.x;
+
+#pragma unroll
+  for (int group_idx = 0; group_idx < num_groups_per_thread; ++group_idx) {
+#pragma unroll
+    for (int element_idx = 0; element_idx < num_consecutive_elements_per_group; ++element_idx) {
+      const auto offset = group_stride * group_idx + element_idx;
+      if (bias_base_idx + offset < bias_size) {
+        dX[input_base_idx + offset] = ComputeGeluGradScalar(
```

Comment:
>input_base_idx + offset [](start = 11, length = 23)

Give a name to this common expression? 
Not sure if compiler will optimize this out....